### ISSUE TITLE
[Feature request] Increase visibility of helix logs for passing tests

### ISSUE THREAD

**Issue by oleksandr-didyk** [dotnet/arcade#14471]
- [ ] This issue is blocking 
- [x] This issue is causing unreasonable pain

Several latest runtime builds have an issue with test artifacts not being present. Specifically, the UI states that there are no artifacts present as the test is not a Helix test, which is not correct.

Example test output: [dev.azure.com](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559787&view=ms.vss-test-web.build-test-results-tab&runId=13352554&resultId=219207&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab)
Example log output: [dev.azure.com](https://dev.azure.com/dnceng-public/public/_build/results?buildId=559787&view=logs&j=52ad279b-b059-569b-870a-4d7b21a81589&t=2ee877e9-a459-5e35-8acf-a38e5009bee3)

**Comment by AlitzelMendez** [comment]
My understanding is that we only provide this information for failing tests, and in this case, we are trying to get the information for passing tests.

This is the code used in this UI: https://github.com/dotnet/arcade-extensions/blob/56b2ea5eae7724cdc048b93e15b241039cbf1bc7/src/helix/index.html#L91

To get the helix logs, we look at the comment of a test, for example: ![image](https://github.com/dotnet/arcade/assets/6895254/0a13bb7f-f340-4814-82da-1778f9f9940a)
 and as it has the helix job id and all the information, we decide it is a helix test and we show the artifacts.

For passing tests, there is no information in there:
![image](https://github.com/dotnet/arcade/assets/6895254/09ee9dc1-b06d-44b2-8a3c-2b0bade16268)
 so we catalogue them as non-helix tests.

I think this it is the expected behavior, no showing the artifacts, but the wording is definitely confusing, as it implies that they are not helix tests, instead of saying that we don’t show information for helix passing tests.

**Comment by ManickaP** [comment]
So how am I supposed to get to the console log from a passing test? In somewhat reasonable fashion.

**Comment by AlitzelMendez** [comment]
@ManickaP you could use: https://helix.dot.net/swagger/ui/index.html#/WorkItem/WorkItem_ListFiles to get the console log file

**Comment by ManickaP** [comment]
And how do I get to that from AzDO pipeline run? Once again, in a reasonable fashion.

**Comment by AlitzelMendez** [comment]
To view the test results from the azure pipeline run perspective, you can follow these steps:
- Find the link to the list of work items that you are interested in from the log output. For example,  [helix.dot.net/api/jobs/2efd3077-377f-49d4-a558-f327b1312cc1/workitems?api-version=2019-06-17](https://helix.dot.net/api/jobs/2efd3077-377f-49d4-a558-f327b1312cc1/workitems?api-version=2019-06-17)
- Search for the test name, such as System.Net.Quic.Functional.Tests, and go to the Details Url. There you will see the link to all the logs.

This is not the most straightforward path, but it is the only one that starts from AzDO.

Alternatively, you can query our Kusto table, which might be easier for you. You can find the documentation here: [KustoDocumentation - Overview (azure.com)](https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/603/KustoDocumentation)

**Comment by ManickaP** [comment]
That's what I was getting at by:
>  in a reasonable fashion.

I know all these steps. I know how to work with Helix API, Kusto etc. I'm trying to point out how painful this whole thing is. Something that should be on the fingertip takes minutes to reach.

See my flow for example:
- Searching leg outputs for job id (finding the right leg is PIA)
  - See how many legs we have: https://dev.azure.com/dnceng-public/public/_build/results?buildId=567528&view=results
  - And browsers find with ctrl+f doesn't work in AzDO because they keep rewriting the content with scroll
  - So you have to find the right leg by slowly scrolling and re-searching what you're looking for
- Then I have to read the output and look for link/job id
  - JobId is GUID, which is nicely unique, but IMHO useless from the user/API perspective - doesn't tell me anything about the job or the run itself, so when I have multiple tabs open with different jobs I get easily lost which one belongs to which run/leg, same applies to links to Console.log
- Now I have the JobId so I can use Kusto/HelixAPI to get to the files, hooray.

Rinse and repeat for each platform I need to check the tests on.

When I complained that passing tests disappeared from Kusto, I was told to use AzDO test result view. So I started using that, no matter how painful it is with is front-end only filtering, slow loading of data and unreliability (if I don't see test there, it still doesn't mean it didn't run). Now this is also useless and I have to comb through leg outputs... I'm so frustrated by this.

Honestly, it's easier to inject a faulty test, let the CI run in vain and burn those $.

I know this is not your fault and no ones in particular, so please don't take my frustration personally. My goal is to point out how hard is to check for test success and make this process easier on us developers.

**Comment by AlitzelMendez** [comment]
Unfortunately, we don’t have this feature of a one-click step for passing tests. I will move this issue to triage in my team so we can discuss whether we can implement this feature, taking into account the number of tests that we run in our system. I will also update the name of the issue to reflect that this is a feature request.

Could you please explain why you need logs for passing tests?

Also, what do you mean by test disappearing from Kusto? Do you mean after 30 days or something else?

Kusto is the centralized place where this information is stored. Based on what you said, I think this query might be useful for you.

```
let buildId = 559787;
let phaseName = "libraries_test_run_release_coreclr_linux_x64_Release";
let testName = "System.Net.Quic.Functional.Tests";
let buildJobsInPhase =
Jobs
| extend p=parse_json(Properties) 
| where tostring(p["BuildId"])  == buildId
| where tostring(p["System.PhaseName"]) == phaseName 
| project JobId;
WorkItems
| where JobId in (buildJobsInPhase)
| where FriendlyName == testName
```

**Comment by ManickaP** [comment]
> why you need logs for passing tests?

Firstly, to see that they ran and passed, mostly on a certain platform(s). It's not straightforward from our CI to see whether the test are being run or not. We had occurrences where we merged seemingly on green to get many reports of broken CI runs right after that because the tests were skipped and we didn't know. We even had gaps in tested platforms of which we didn't know.
Secondly, S.N.Quic is opt-in, meaning if there is MsQuic on the machine, it will light up. So seeing test suite ran and passed is not enough as we need to check whether the individual tests ran or were skipped. 
Lastly, we log the version of MsQuic in our tests, which once again helps seeing whether we've updated the machines the latest.

>  test disappearing from Kusto

Some time ago, before the big Kusto change of tables, TestResults contained successful runs as well. So it was much easier to check for passing tests. Figure out where the tests are actually running, how many tests are passing in comparison to failures etc.

Right know it's indistinguishable between test passing and test not running at all.
